packs/chef/actions/lib/shell_out.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python2.7
+
+import sys
+import os
+import subprocess
+import select
+
+# Starts command in subprocess processing output as it appears.
+def shell_out(command, env={}):
+    env = os.environ.copy().update(env)
+    proc = subprocess.Popen(args=command, stdin=None, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, env=env)
+    while True:
+        fdlist = [proc.stdout.fileno(), proc.stderr.fileno()]
+        ios = select.select(fdlist, [], [])
+        for fd in ios[0]:
+            if fd == proc.stdout.fileno():
+                sys.stdout.write(proc.stdout.read())
```

Comment:
Yeah, I think readline should work.

Also, it would be nice to make this function a bit more generic (allow user to pass in streams / file-like objects to which the stdout and stderr will be written) and move it to https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/shell.py